engines/saga/font.cpp

```diff
@@ -653,17 +700,69 @@ void DefaultFont::outFont(const FontStyle &drawFont, const char *text, size_t co
 	ct = count;
 
 	bool isBig5 = !!_chineseFont;
+	bool isJohab = !!_koreanFont;
 
 	// Draw string one character at a time, maximum of 'draw_str'_ct
 	// characters, or no limit if 'draw_str_ct' is 0
 	for (; *textPointer && (!count || ct); textPointer++, ct--) {
-		int charWidth;
-		int charHeight;
-		int charTracking;
-		int rowLength;
-		const byte *bitmap;
 		c_code = *textPointer & 0xFFU;
 
+		if ((c_code & 0x80) && isJohab) {
+			byte leading = c_code;
+			byte trailing = *++textPointer & 0xFFU;
+			ct--;
+			if (ct == 0 || trailing == 0)
+				break;
+			uint16 full = ((leading & 0x7f) << 8) | trailing;
+			int initial = (full >> 10) & 0x1f;
+			int mid = (full >> 5) & 0x1f;
+			int fin = full & 0x1f;
+			int initidx = initial - 1;
+			static const int mididxlut[0x20] = {
+				-1, -1, 0, 1, 2, 3, 4, 5, -1, -1, 6, 7, 8, 9,
+				10, 11, -1, -1, 12, 13, 14, 15, 16, 17, -1, -1,
+				18, 19, 20, 21, -1, -1};
+			// TODO: check this.
+			static const int mid2inivariant[32] = {
+				0, 0, 1, 0, 0, 0, 0, 0,
+				0, 0, 0, 0, 0, 1, 3, 3,
+				0, 0, 3, 1, 1, 3, 3, 3,
+				0, 0, 1, 1, 3, 0, 0, 0,
+			};
+			int mididx = mididxlut[mid];
+			int finidx = fin >= 0x12 ? fin - 2 : fin - 1;
+			if (initial >= 0x15 || initidx < 0 || mididx < 0 || fin == 0 || fin == 0x12 || fin >= 0x1e) {
+				// TODO: non-jamo
```

Comment:
Jamo fonts are used in Korean. What's this for? And what needs to be done here?

Reply:
Korean has also some non-Hangul characters like Hanja, pictograms and pseudographics. Compare it to English using digits, emojis and pseudographics

Reply:
Clarified